src/Illuminate/Hashing/Argon2IdHasher.php

```diff
@@ -18,13 +18,13 @@ class Argon2IdHasher extends ArgonHasher
      */
     public function check(#[\SensitiveParameter] $value, $hashedValue, array $options = [])
     {
```

Comment:
seems like a logic error to check for the hash compatibility before checking if the hashed value is even non-null or empty. 

the original if statement would never be reached. contrast to the other hashing implementations that were missing this entirely

```
if (is_null($hashedValue) || strlen($hashedValue) === 0) {
            return false;
        }
```